tests/consumption_test.cpp

```diff
@@ -0,0 +1,72 @@
+#include "catch/catch.hpp"
+
+#include "avatar.h"
+#include "bionics.h"
+#include "consumption.h"
+#include "effect.h"
+#include "item.h"
+#include "mutation.h"
+
+namespace consumption_test
+{
+
+static const bionic_id bio_digestion( "bio_digestion" );
+static const efftype_id effect_foodpoison( "foodpoison" );
+static const trait_id trait_POISRESIST( "POISRESIST" );
+static const trait_id trait_EATDEAD( "EATDEAD" );
+
+// Doesn't test the chance to get "real" poisoned; that depends on deterministically
+// manipulating RNG and I don't want to figure that out yet.
+TEST_CASE( "consume_poison", "[consumption]" )
+{
+    item test_poison{};
+    avatar test_consumer{};
+
+    SECTION( "not poison" ) {
+        test_poison.poison = 0;
+        consume_poison( test_consumer, test_poison );
+        effect &is_poisoned = test_consumer.get_effect( effect_foodpoison );
```

Comment:
You are getting a reference and aren't changing anything about the stored effect. This should be

```suggestion
        const effect &is_poisoned = test_consumer.get_effect( effect_foodpoison );
```